probe-rs/src/architecture/arm/sequences/nxp_armv8m.rs

```diff
@@ -19,6 +20,346 @@ use crate::{
+            let value = interface.read_word_32(0x40034FE0)?;
+
+            if (value & 0x4) == 0x4 {
+                timeout = false;
+                break;
+            }
+        }
+
+        if timeout {
+            tracing::warn!("Failed: Wait for flash word read to finish");
+            return Err(ArmError::Timeout);
+        }
+
+        if (interface.read_word_32(0x4003_4fe0)? & 0xB) == 0 {
+            tracing::info!("No Error reading Flash Word with Reset Vector");
+
+            if (interface.read_word_32(0x400a_cffc)? & 0xC != 0x8)
+                || (interface.read_word_32(0x400a_cff8)? & 0xC != 0x8)
+            {
+                // ENABLE_SECURE_CHECKING is set to restrictive mode, access secure addresses
+                reset_vector_addr = 0x10000004;
+            }
+
+            reset_vector = interface.read_word_32(reset_vector_addr)?;
+        }
+
+        if reset_vector != 0xffff_ffff {
+            tracing::info!("Breakpoint on user application reset vector: {reset_vector:#010x}");
+
+            interface.write_word_32(0xE000_2008, reset_vector | 1)?;
+            interface.write_word_32(0xE000_2000, 3)?;
+        }
+
+        if reset_vector == 0xffff_ffff {
+            tracing::info!("Enable reset vector catch");
+
+            let mut demcr = Demcr(interface.read_word_32(Demcr::get_mmio_address())?);
+
+            demcr.set_vc_corereset(true);
+
+            interface.write_word_32(Demcr::get_mmio_address(), demcr.into())?;
+        }
+
+        let _ = interface.read_word_32(Dhcsr::get_mmio_address())?;
+
+        tracing::debug!("reset_catch_set -- done");
+
+        Ok(())
+    }
+
+    fn reset_catch_clear(
+        &self,
+        interface: &mut dyn ArmProbe,
+        _core_type: crate::CoreType,
+        _debug_base: Option<u64>,
+    ) -> Result<(), ArmError> {
+        interface.write_word_32(0xE000_2008, 0x0)?;
+        interface.write_word_32(0xE000_2000, 0x2)?;
+
+        let mut demcr = Demcr(interface.read_word_32(Demcr::get_mmio_address())?);
+
+        demcr.set_vc_corereset(false);
+
+        interface.write_word_32(Demcr::get_mmio_address(), demcr.into())
+    }
+
+    fn reset_system(
+        &self,
+        interface: &mut dyn ArmProbe,
+        _core_type: crate::CoreType,
+        _debug_base: Option<u64>,
+    ) -> Result<(), ArmError> {
+        let mut aircr = Aircr(0);
+        aircr.vectkey();
+        aircr.set_sysresetreq(true);
+
+        let mut result = interface.write_word_32(Aircr::get_mmio_address(), aircr.into());
+
+        if result.is_ok() {
+            result = interface.flush();
+        }
+
+        if let Err(e) = result {
+            tracing::warn!("Error requesting reset: {:?}", e);
+        }
+
+        tracing::info!("Waiting after reset");
+        thread::sleep(Duration::from_millis(10));
+
+        let start = Instant::now();
+
+        let mut timeout = true;
+
+        while start.elapsed() < Duration::from_micros(50_0000) {
```

Comment:
```suggestion
        while start.elapsed() < Duration::from_millis(500) {
```

This was a bit of a misleading grouping, it wasn't clear if the intent was 50 or 500ms.